Group_id and groups type

tschmidtb51 opened the pull request:
- resolves #57 
- add "group_id_t" and "groups_t"
- refactor to use new types
- add generic keywords for documentation

tschmidtb51 committed:
Group_id and groups type

- resolves oasis-tcs/csaf#57
- add "group_id_t"
- refactor usage of group_t to new "group_id_t"
- add generic keywords "title", "description", "examples" to the definition

tschmidtb51 committed:
Group_id and groups type

- resolves second part of oasis-tcs/csaf#57
- add "groups_t"
- refactor usage of groups_t to new "groups_t"
- add generic keywords "title", "description" to the definition

santosomar commented on csaf_2.0/json_schema/csaf_json_schema.json:
```diff
@@ -107,6 +107,22 @@
+      },
+      "uniqueItems": true
+    },
+    "group_id_t": {
```
During the June 24, 2020 monthly meeting, it was suggested and discussed to rename to `product_group_id_t` to make relationship to products explicit, as stated by @tolim above. The TC decided that this is the right approached (i.e., rename to `product_group_id_t`).

tschmidtb51 replied:
Addressed in [https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830](https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830)

santosomar reviewed (changes requested):
During the June 24, 2020 monthly meeting, it was suggested and discussed to rename to product_group_id_t to make relationship to products explicit, as stated by @tolim above. The TC decided that this is the right approached (i.e., rename to product_group_id_t).

tschmidtb51 committed:
Group_id and groups type

- addresses discussion raised in TC meeting (documented in oasis-tcs/csaf#58)
- change "groups_t" to "product_groups_t"
- change "group_id_t" to "product_group_id_t"

tschmidtb51 committed:
Merge branch 'master' into group_id-groups-type

- resolves conflict in csaf_2.0/json_schema/csaf_json_schema.json

tschmidtb51 commented on csaf_2.0/json_schema/csaf_json_schema.json:
```diff
@@ -107,6 +107,22 @@
         }
       }
     },
+    "groups_t": {
```
Addressed in [https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830](https://github.com/oasis-tcs/csaf/pull/58/commits/2a61d238696fd2747ea678081a7fedf4a4010830).

santosomar reviewed (approved):
Thank you for taking care of this so quickly.

sthagen commented on csaf_2.0/json_schema/csaf_json_schema.json:
```diff
@@ -708,7 +723,7 @@
                 },
                 "group_ids": {
-                  "$ref": "#/definitions/products_t"
+                  "$ref": "#/definitions/product_groups_t"
```
:+1:
